### ISSUE TITLE
Full testing suite overview

### ISSUE THREAD

**Issue by cedricdcc** [vliz-be-opsci/py-RDF-store#27]
### Integration-testing

- [x] : create_rdf_store with 0 arguments to see if result is MEMORYSTORE
- [x] : create_rdf_store with 1/2 arguments to see if result is URISTORE
- [x] : create_rdf_store with <3 arguments to see if assertion is given

### RDFStore-testing

For both the MEMORYSTORE  and URISTORE

- [ ] : check if .select() returns a rdflib.Result object for a good formulated SPARQL query
- [ ] : check if .select() return an assertion for bad SPARQL given
- [ ] : check if named_graph given the info only comes from named graph
- [ ] : check insert() without named graph given is in context urn:_nil
- [ ] : check insert() with named graph given is in right context 
- [ ] : check if insert() discards blank nodes from given graph
- [ ] : check if verify_max_age() gives back the right age for non context and context given
- [ ] : check lastmod_timestamp() gives back right time
- [ ] : check if forget_graph() does not delete a graph but deletes it from the admin_context
- [ ] : check if drop graph deletes grapg content and in admin context

### Edgecases

*all edgecases here*

**Comment by cedricdcc** [comment]
@marc-portier any other tests in mind?

**Comment by marc-portier** [comment]
not sure about some of these:
- check if insert() discards blank nodes from given graph --> what do you mean with discard?

and I think these are already in there in some way:
- check if .select() returns a rdflib.Result object for a good formulated SPARQL query
- check lastmod_timestamp() gives back right time
- check if verify_max_age() gives back the right age for non context and context given
- check if forget_graph() does not delete a graph but deletes it from the admin_context

leaving  these to be done:
- check if .select() return an assertion for bad SPARQL given
- check insert() without named graph given is in context urn:_nil
- check if named_graph given the info only comes from named graph
- check insert() with named graph given is in right context --> but can´t be disjunct from the one above IMHO
- check if drop graph deletes graph content and in admin context

also, we should use the coverage report to decide to include more and better tests !

**Comment by marc-portier** [comment]
considering this again for upcoming 0.0.3 

but have my reservations about some of the demanded tests (e.g. the one about urn:_nil) 

I think unit tests should test the contracts of the service on offer, not the internals of its implementation (on this case: the outside client does not need to know about urn:_nil so has no business testing it, specially since distinct implementations could have taken other options in this regard too) 

I'd rather let us be driven by a desire to purely increase test-coverage ...

current results of that:
```
---------- coverage: platform linux, python 3.10.12-final-0 ----------
Name                     Stmts   Miss  Cover   Missing
------------------------------------------------------
pyrdfstore/__init__.py       4      0   100%
pyrdfstore/build.py          9      0   100%
pyrdfstore/clean.py         94      3    97%   243, 248-249
pyrdfstore/store.py        187     21    89%   74, 129-130, 167, 177-178, 190-191, 206, 220, 269, 280, 294, 312, 339-342, 530, 533, 536, 540
------------------------------------------------------
TOTAL                      294     24    92%
```

**Comment by marc-portier** [comment]
working on it inside #53 
coverage on cleaning now bumped (and discovered a lingering issue) to
```
---------- coverage: platform linux, python 3.10.12-final-0 ----------
Name                     Stmts   Miss  Cover   Missing
------------------------------------------------------
pyrdfstore/__init__.py       4      0   100%
pyrdfstore/build.py          9      0   100%
pyrdfstore/clean.py         97      0   100%
pyrdfstore/store.py        187     21    89%   74, 129-130, 167, 177-178, 190-191, 206, 220, 269, 280, 294, 312, 339-342, 530, 533, 536, 540
------------------------------------------------------
TOTAL                      297     21    93%
```

**Comment by marc-portier** [comment]
ending up at
```
---------- coverage: platform linux, python 3.10.12-final-0 ----------
Name                     Stmts   Miss  Cover   Missing
------------------------------------------------------
pyrdfstore/__init__.py       4      0   100%
pyrdfstore/build.py          9      0   100%
pyrdfstore/clean.py         97      0   100%
pyrdfstore/store.py        179      3    98%   335-338
------------------------------------------------------
TOTAL                      289      3    99%
```

leaving out:
- 335-338 are about not having a write_uri rendering a read-only store 
¯\\\_(ツ)\_/¯

which currently feels as odd an exceptional anyway?

**Comment by marc-portier** [comment]
closed in merge of #53
